x/gamm/keeper/pool.go

```diff
@@ -100,6 +143,13 @@ func (k Keeper) setPool(ctx sdk.Context, pool swaproutertypes.PoolI) error {
 	return nil
 }
 
+// initializePoolId initializes pool id to 0.
+func (k Keeper) initializePoolId(ctx sdk.Context) {
+	store := ctx.KVStore(k.storeKey)
+	poolId := &gogotypes.UInt64Value{Value: 0}
+	osmoutils.MustSet(store, types.KeyNextGlobalPoolId, poolId)
+}
```

Comment:
This should be setting to 1, right?

Reply:
Yes based on: https://github.com/osmosis-labs/osmosis/blob/5d54522a0dfa94f4f3b1664e7fa8fa183abe5da5/x/gamm/types/genesis.go#L11

However, that index is only kept for performing migrations in the upgrade handler and is to be removed in the upgrade after the migrations are done. It is not to be used going forward. So the final value should not matter much